seshdash/views.py

```diff
@@ -796,62 +797,81 @@ def historical_data(request):
         context_dict['sort_dict'] = sort_data_dict
         return render(request, 'seshdash/data_analysis/historical-data.html', context_dict);
 
-#function for Graph Generations
+
+@require_GET
 @login_required
-def graphs(request):
+def graphs(request): 
+    """
+    Returns json, containing data that is used in data analysis graphs
+    """
+    results = []
 
-    # if ajax request
-    if request.method == 'POST':
+    # Getting values from Post request
+    time = request.GET.get('time', '') # This is the time range it has to be: 24h, 7d or 30d
+    choices = request.GET.getlist('choice[]')
+    active_id = request.GET.get('active_site_id', None)
+    start_time = request.GET.get('start_time', None)
+    end_time = request.GET.get('end_time', None)
+    resolution = request.GET.get('resolution', '1h')
+    current_site = Sesh_Site.objects.filter(id=active_id).first()
+    
+
+    print "Got values %s %s %s of site %s " % (time, choices, active_id, current_site)
+    if not current_site:
+        return HttpResponseBadRequest("Invalid site id, No site was found for the given site id")
```

Comment:
we  should also check the user has permission to view the site.
Else someone can send a GET request for a site the don't have permission to view and still get data.